archinstall/lib/args.py

```diff
@@ -211,7 +211,15 @@ def from_config(cls, args_config: dict[str, Any], args: Arguments) -> 'ArchConfi
 		if parallel_downloads := args_config.get('parallel_downloads', 0):
 			arch_config.parallel_downloads = parallel_downloads
 
-		arch_config.swap = args_config.get('swap', True)
+		# Parse swap config - transform bool/dict into ZramConfiguration
+		swap_arg = args_config.get('swap', True)
+		if isinstance(swap_arg, bool):
+			arch_config.swap = ZramConfiguration(enabled=swap_arg)
+		elif isinstance(swap_arg, dict):
+			algo = swap_arg.get('algo', ZramAlgorithm.ZSTD.value)
```

Comment:
Can we move this into a `parse_args` function like all the others, example https://github.com/archlinux/archinstall/blob/5fcea379b9cd1f65343056dcfd70a441dc0ab744/archinstall/lib/models/application.py?plain=1#L62